### PR TITLE
Enabling vector support on Azure PostgreSQL Flexible Server

### PR DESCRIPTION
**Pull request by aaronpowell**
Initial prototype for #3052

Since the CDK doesn't currently support configuring extensions on the PG Server resource (see [this issue](https://github.com/Azure/azure-sdk-for-net/issues/43175)), we have to generate the Bicep on the fly to do that. There's no easy hook that can be used after the original call to PublishAsAzurePostgresFlexibleServer to have it output values from the bicep, so instead we have to provide our own configureResource method.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4335)

### CODE REVIEW

**Comment by davidfowl** [comment]
We would change the CDK to support this.

File: src/Aspire.Hosting.Azure.PostgreSQL/AzurePostgresExtensions.cs [hunk 1]
```diff
@@ -171,4 +171,42 @@ public static class AzurePostgresExtensions
...
+      properties: {
+        value: 'VECTOR'
+        source: 'user-override'
+      }
+    }
+    """);
+
+        builder.AsAzurePostgresFlexibleServer((resource, construct, server) =>
+        {
+            construct.AddOutput(server.AddOutput("name", data => data.Name));
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure.PostgreSQL/AzurePostgresExtensions.cs:206]: This does what you want:

```suggestion
            server.AddOutput("name", data => data.Name);
```

File: src/Aspire.Hosting.Azure.PostgreSQL/AzurePostgresExtensions.cs [hunk 2]
```diff
@@ -171,4 +171,42 @@ public static class AzurePostgresExtensions
...
+            template.WithParameter("postgresServerName", resource.GetOutput("name"));
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure.PostgreSQL/AzurePostgresExtensions.cs:207]: This being in the callback is quite strange but I see why you have to do it that way. @mitchdenny another strike against this pattern.

Comment (mitchdenny) [reply src/Aspire.Hosting.Azure.PostgreSQL/AzurePostgresExtensions.cs:207]: out params :) /jk

**Comment by aaronpowell** [comment]
> We would change the CDK to support this.

I do have an open PR for this on the CDK - https://github.com/Azure/azure-sdk-for-net/pull/44315

But I also wanted to look at how to unblock people who might want it sooner than those pieces become available.

**Comment by davidfowl** [comment]
I understand but we're not going to take the non CDK version in this repo. That can be a sample.

**Comment by eerhardt** [comment]
Is this PR stale? Can it be closed? Or are we going to take this change?

**Comment by aaronpowell** [comment]
Yep, I'll close it.

I have a PR open on the CDK to get it supported in the .NET API - https://github.com/Azure/azure-sdk-for-net/pull/44315

**Pull request closed without merging**